### DISCUSSION TITLE
加入符号对照表和缩略语对照表后，更新文档出现错误

### DISCUSSION THREAD

**Discussion by luoladala** [note286/xduts#72] [Q&A] [answered]
正文代码中加入

```latex
style / customize-los = false,
 style / customize-loa = false,
style / colspec-los   = {>{\hskip1cm}Q[l,h]<{\hskip2cm}X[l,h]},
 style / colspec-loa   = {>{\hskip1cm}Q[l,h]<{\hskip1cm}X[-1,l,h]X[-1,l,h]},
info / los = los.tex,
info / loa = loa.tex,
```

语句后，在命令行执行`tlmgr update --all --self`进行更新后出错
错误为：

```
! LaTeX3 Error: Mismatched LaTeX support files detected.
(LaTeX3)        Loading 'expl3.sty' aborted!
(LaTeX3)        
(LaTeX3)        The L3 programming layer in the LaTeX format
(LaTeX3)        is dated 2022-02-24, but in your TeX tree the files require
(LaTeX3)        at least 2023-02-22.
```

更换编译器错误显示为：

```
LaTeX3 Error: Mismatched LaTeX support files detected. \ExplLoaderFileDate{expl3.sty}}
Undefined control sequence. \sys_if_output_pdf:TF
LaTeX3 Error: Control sequence \ifctexpdf already defined. { \cs_new_eq:NN \ifctexpdf
```

查找网站后使用如下方法：
命令行执行`fmtutil-user --all`和`fmtutil-sys --all`依然报错未解决，
请问一下大佬该问题如何解决？万分感谢

**Comment by note286** [comment, marked as answer]
请参考[expl3.sty:77: LaTeX3 Error: Mismatched LaTeX support files detected. · Issue #811 · latex3/latex3](https://github.com/latex3/latex3/issues/811)。

**Reply by luoladala** [reply]
1. 您好！已经按照所有方法执行：运行过kpsewhich --engine=pdftex --all pdflatex.fmt语句，也运行过fmtutil-user --all和fmtutil-sys --all语句，依然报错：LaTeX3 Error: Mismatched LaTeX support files detected. \ExplLoaderFileDate{expl3.sty}}

Undefined control sequence. \sys_if_output_pdf:TF
LaTeX3 Error: Control sequence \ifctexpdf already defined. { \cs_new_eq:NN \ifctexpdf
仍然未解决。感觉搜索了很多方法都是运行fmtutil-user --all和fmtutil-sys --all语句，可我这个没有用，
不知道问题出在哪里？

**Reply by note286** [reply]
如果还有问题，可以尝试重新安装。

**Reply by note286** [reply]
你可以看看 [编译失败，\sys\_if\_output\_pdf:TF 未定义 · Issue #149 · CTeX-org/forum](https://github.com/CTeX-org/forum/issues/149)

**Reply by note286** [reply]
或者你选择使用MiKTeX也可以，安装更快。

**Reply by luoladala** [reply]
您好！我还想浅问一下，我在命令行执行了fmtutil-user --all语句，但是最后出现如下一些报错：
No pages of output.
Transcript written on lollipop.log.
fmtutil [INFO]: log file copied to: c:/users/wu/.texlive2022/texmf-var/web2c/tex/lollipop.log
fmtutil [INFO]: c:/users/wu/.texlive2022/texmf-var/web2c/tex/lollipop.fmt installed.
fmtutil [ERROR]: running `eptex -ini   -jobname=eptex -progname=eptex *eptex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `euptex -ini   -jobname=euptex -progname=euptex *euptex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=pdftex -progname=pdftex -translate-file=cp227.tcx *pdfetex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: no (or empty) hitex.fmt made by: hitex -ini   -jobname=hitex -progname=hitex -etex -ltx hitex.ini <nul
fmtutil [ERROR]: running `xetex -ini   -jobname=xetex -progname=xetex -etex xetex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `xetex -ini   -jobname=xelatex-dev -progname=xelatex-dev -etex xelatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `euptex -ini   -jobname=uplatex -progname=uplatex *uplatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=latex -progname=latex -translate-file=cp227.tcx *latex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `eptex -ini   -jobname=platex -progname=platex *platex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=eplain -progname=eplain -translate-file=cp227.tcx *eplain.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=jadetex -progname=jadetex *jadetex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=pdflatex-dev -progname=pdflatex-dev -translate-file=cp227.tcx *pdflatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=pdflatex -progname=pdflatex -translate-file=cp227.tcx *pdflatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `eptex -ini   -jobname=platex-dev -progname=platex-dev *platex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=mllatex -progname=mllatex -translate-file=cp227.tcx -mltex *mllatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=xmltex -progname=xmltex *xmltex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=etex -progname=etex -translate-file=cp227.tcx *etex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=pdfjadetex -progname=pdfjadetex *pdfjadetex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `euptex -ini   -jobname=uplatex-dev -progname=uplatex-dev *uplatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=latex-dev -progname=latex-dev -translate-file=cp227.tcx *latex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `xetex -ini   -jobname=xelatex -progname=xelatex -etex xelatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=pdfetex -progname=pdfetex -translate-file=cp227.tcx *pdfetex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=pdfxmltex -progname=pdfxmltex *pdfxmltex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [INFO]: disabled formats: 6
fmtutil [INFO]: successfully rebuilt formats: 32
fmtutil [INFO]: failed to build: 23 (eptex/eptex euptex/euptex pdftex/pdftex hitex/hitex xetex/xetex xetex/xelatex-dev euptex/uplatex pdftex/latex eptex/platex pdftex/eplain pdftex/jadetex pdftex/pdflatex-dev pdftex/pdflatex eptex/platex-dev pdftex/mllatex pdftex/xmltex pdftex/etex pdftex/pdfjadetex euptex/uplatex-dev pdftex/latex-dev xetex/xelatex pdftex/pdfetex pdftex/pdfxmltex)
fmtutil [INFO]: total formats: 61
fmtutil [INFO]: exiting with status 23
D:\LaTeX\texlive\2022\bin\win32\runscript.tlu:915: command failed with exit code 23:
perl.exe d:\LaTeX\texlive\2022\texmf-dist\scripts\texlive\fmtutil.pl  --user --all
以上
请问这些信息中是否能解释我运行此命令为何仍然报错吗？

**Reply by note286** [reply]
你就用这么一回了，没必要纠结这些了。

**Reply by luoladala** [reply]
好滴！谢谢大佬了，我再重装吧估计就哈哈哈哈唉
